### PR TITLE
Mock out test interactions with swift.org and github.com

### PR DESCRIPTION
**Pull request by cmcgee1024**
The unit tests currently perform numerous requests against these websites, which causes problems. First, GitHub API's are rate limited, which is often encountered while running the tests both in CI and at desk since not all environments are prepared to manage their own app/dev token. Downloads from swift.org, usually toolchains, are large, take a significant time for each to download in aggregate during testing. Ultimately, network interactions can fail due to network conditions outside of the control of the tests.

There is already the infrastructure in the tests to mock the downloading of toolchains by locally constructing them with executable shell scripts in place of the real tools. In the case of macOS there is also logic to assemble the .pkg file. Use this mock for the majority of the unit tests.

Create a small set of integration and e2e tests that do make real network requests.

Create static files that represent a mocked snapshot of the GitHub API's response JSON and use those when using the mock toolchain downloader during the automated tests. Add these as resources to the test target so that they can be loaded at the runtime of the tests.

In order to decrease the exposure to network issues related to the Ubuntu keyserver in the automated tests add a latch so that the GPG keys are only refreshed once per session of the swiftly executable. Since swiftly operates normally as a short-lived process this should not affect the real-world usage of it, but it makes a big improvement in both the speed and reliability of the automated tests. It's very unlikely that in the span of a test run that the swift keys will need to be refreshed.

### CODE REVIEW

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

**Comment by cmcgee1024** [comment]
@swift-server-bot test this please

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

**Review by adam-fowler — APPROVED** [review]
This is great. Nice to see all the tests complete in such a short time.
Maybe not for this PR but the system relies a lot on global variables which means there is no chance we can run these tests in parallel. Would be good to look in the future at cleaning up these globals. 
Any attempt to support strict concurrency will probably require some work to these globals.

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

File: Tests/SwiftlyTests/SwiftlyTests.swift [hunk 1]
```diff
@@ -451,30 +468,139 @@ private struct MockHTTPRequestExecutor: HTTPRequestExecutor {
...
+            PackageResources.swift_tags_page25_json
+        case "26":
+            PackageResources.swift_tags_page26_json
+        case "27":
+            PackageResources.swift_tags_page27_json
+        default:
+            Array("[]".utf8)
+        }
+
+        return HTTPClientResponse(body: .bytes(ByteBuffer(data: Data(payload))))
```
[the remark this answers is not in the document]
Comment (cmcgee1024) [reply Tests/SwiftlyTests/SwiftlyTests.swift]: Thanks, I've fixed this in the latest commit.

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

**Pull request merged**